python/ray/ml/train/integrations/tensorflow/tensorflow_trainer.py

```diff
@@ -0,0 +1,54 @@
+from typing import Callable, Optional, Dict
+
+from ray.train.tensorflow import TensorflowConfig
+from ray.ml.trainer import GenDataset
+from ray.ml.train.impl.data_parallel_trainer import DataParallelTrainer
+from ray.ml.config import ScalingConfig, RunConfig
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+from ray.util import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class TensorflowTrainer(DataParallelTrainer):
+    """A Trainer for data parallel TensorFlow training.
+
+    This Trainer runs the provided function on multiple Ray Actors. The
+    provided datasets are automatically sharded to each Ray actor running
+    the training function.
+
+    Args:
+        train_func (Callable): The training function to execute.
+            This can either take in no arguments or a ``config`` dict.
+        train_func_config (Optional[Dict]): Configurations to pass into
+            ``train_func`` if it accepts an argument.
+        tensorflow_config (Optional[TensorflowConfig]): Configuration
```

Comment:
remove typing?

Reply:
It's still present in https://github.com/ray-project/ray/pull/22989/

Reply:
Will do!